### PR TITLE
#Issue30-Added functionality to merge pdf files

### LINKED ISSUE #11: As a JavaFX developer, I want doxygen formatted code, so that the design document can be generated easily whenever needed

### LINKED ISSUE #19: As a JavaFX user, I want merge and split functionalities, so that the user can either add or split the pdf

### LINKED ISSUE #30: As part of Issue19, I want to add Merge functionality to the pdf file.

### LINKED ISSUE #41: As part of Issue 19, add an alert message, when merge or split is done successfully

### PR DESCRIPTION
**Pull request by paidikondala**
Added functionality to merge pdf files.
Also added test cases for split and merge functionalities

### CODE REVIEW

**Review by aanuhya31 — APPROVED** [review]
The code looks fine

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		    pagination.setPageFactory(index -> {
+		    	System.out.println("haiinpagena");
+	            ImageView currentPageImageView = new ImageView(model.getImage(0));
+	            pageImageGroup.getChildren().clear();
+	            pageImageGroup.getChildren().add(currentPageImageView);
+	   
+				return pageImageGroup;
+	        });
+	        File f = new File(current);
+			//when(fileChooser.showOpenDialog(any()).thenReturn(fileChooser);
```
Comment (sireeshanalabothu) [inline src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: remove the commented lines of code if not necessary

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+				@Override
+				public void run() {
+					// TODO Auto-generated method stub
+					try {
+						new SplitAndMerge().splitter("1",pagination,model);
+					} catch (Exception e) {
+						// TODO Auto-generated catch block
+						e.printStackTrace();
+					}
+
```
Comment (sireeshanalabothu) [inline src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: remove the more no of empty lines and format the code

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		AIRViewerModel model = new AIRViewerModel(exist);
+		pagination.setPageCount(model.numPages());
+	    pagination.setDisable(true);
+	    Group pageImageGroup = new Group();
+	    pagination.setPageFactory(index -> {
+	    	System.out.println("haiinpagena");
+            ImageView currentPageImageView = new ImageView(model.getImage(0));
+            pageImageGroup.getChildren().clear();
+            pageImageGroup.getChildren().add(currentPageImageView);
+
```
Comment (sireeshanalabothu) [inline src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: code lines are not structured. plrase align

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java (hunk 1 above)
Comment (paidikondala) [reply src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: Removed commented lines

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java (hunk 2 above)
Comment (paidikondala) [reply src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: Removed the empty lines and formatted code. Please review it

File: src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java (hunk 3 above)
Comment (paidikondala) [reply src/test/java/edu/wright/airviewer2/SplitAndMergeTest.java]: Formatted the code and committed the changes

**Review by sireeshanalabothu — APPROVED** [review]

**Pull request merged**